### PR TITLE
Document ready rule

### PR DESCRIPTION
**Pull request by awisema**
Adds new rule:
image

The client must have files with all of the the specified tags to match the MUST case.

The tags are added to the list of file tags that will appear in the match files list, and are marked with a `*`:
image

### CODE REVIEW

File: app/models/client_opportunity_match.rb [hunk 1]
```diff
@@ -884,6 +884,10 @@ def similar_matches
       end
   end
 
+  def associated_file_tags
+    sub_program.file_tags.pluck(:tag_id) + requirements_with_inherited.map(&:associated_file_tags).flatten.compact
```
Comment (eanders) [inline app/models/client_opportunity_match.rb]: I wonder if it makes sense to expose which thing is requiring a particular tag?  As we do when we expose the rules and expose the source of the requirement. Something like:

State-issued id _(program name)_
DD-214 _(Vet Funding Source)_

File: app/models/rule.rb [hunk 2]
```diff
@@ -39,12 +41,15 @@ def variable_requirement?
...
 
-  def display_for_variable value
+  def display_for_variable(_value)
     nil
   end
 
   def apply_to_match(match)
   end
+
+  def associated_file_tags(value)
```
Comment (eanders) [inline app/models/rule.rb]: should this return an empty array by default?
```suggestion
  def associated_file_tags(value)
    []
```

**Review by eanders — CHANGES_REQUESTED** [review]
This looks good, just a couple of questions, and one request to add the source to the display on the match.  Thanks!

File: app/models/rule.rb (hunk 2 above)
Comment (awisema) [reply app/models/rule.rb]: Doesn't really matter as the `nil`s get cleaned up at the end -- but this makes the API more consistent, and I think that is a good thing.

File: app/models/client_opportunity_match.rb (hunk 1 above)
Comment (awisema) [reply app/models/client_opportunity_match.rb]: Like so?
image

Comment (eanders) [reply app/models/client_opportunity_match.rb]: Exactly.  Thanks!

**Pull request merged**